cmd/juju/application/bundle.go

```diff
@@ -1105,44 +1105,54 @@ func processValue(baseDir string, v interface{}) (interface{}, bool, error) {
 	return result, true, nil
 }
 
-type bundleConfig struct {
-	Applications map[string]*charm.ApplicationSpec `yaml:"applications"`
-	// TODO soon, add machine mapping and space mapping.
-}
-
 type bundleConfigValueExists struct {
 	Applications map[string]map[string]interface{} `yaml:"applications"`
 }
 
-func processBundleConfig(data *charm.BundleData, bundleConfigFile string) error {
-	if bundleConfigFile == "" {
-		// Nothing to do here.
-		return nil
-	}
-
-	bundleConfigFile, err := utils.NormalizePath(bundleConfigFile)
-	if err != nil {
-		return errors.Annotate(err, "unable to normalise bundle-config file")
-	}
-	// Make sure the filename is absolute.
-	if !filepath.IsAbs(bundleConfigFile) {
-		cwd, err := os.Getwd()
+func processBundleConfig(data *charm.BundleData, bundleConfigFiles ...string) error {
+	for _, filename := range bundleConfigFiles {
+		bundleConfigFile, err := utils.NormalizePath(filename)
 		if err != nil {
+			return errors.Annotate(err, "unable to normalise bundle-config file")
```

Comment:
Could we please add a filename into the error message?

Reply:
The filename is already in the text from NormalizePath.